### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by abelsiqueira**

### CODE REVIEW

**Comment by abelsiqueira** [comment]
@dpo, do you want to review this? It's the beginning of tests. Hopefully it'll be extended to equality-constrained soon.

- Removes extra `kwargs...` from solvers;
- Add consistency checks, that tests whether the solvers accept `atol`, `rtol`, `max_eval`, and `max_time`;
- Test that `max_eval=1` leads to `:max_eval` status, and same for `max_time`;
- Adds `test/solvers/unconstrained.jl`, which are basic problems that solvers for unconstrained problems should handle;
- Same for `bound-constrained.jl` and `unconstrained-nls`;
- These were created from early tests for `tron` and `trunkls`, so most of the specific tests for these solvers were removed;
- `trunkls` still has a few specific tests;
- And finally, I only test `trunkls` with `cgls`, because there are multiprecision tests.

File: test/solvers/unconstrained.jl [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        nlp = CUTEstModel(p)
+        stats = with_logger(NullLogger()) do
+          solver(nlp, max_time=3.0)
+        end
+        finalize(nlp)
+
+        ctype = "unc"
+        stline = statsline(stats, [:objective, :dual_feas, :elapsed_time, :iter, :neval_obj,
+                                   :neval_grad, :neval_hprod, :status])
+        @info @sprintf("%8s  %5d  %4s  %s\n", p, nlp.meta.nvar, ctype, stline)
```
Comment (dpo) [inline test/solvers/unconstrained.jl]: Is this necessary? What is it testing?

File: test/solvers/unconstrained-nls.jl [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+      @test stats.status == :first_order
+    end
+  end
+
+  @testset "Extended Rosenbrock" begin
+    n = 30
+    nls = ADNLSModel(x->[[10 * (x[i+1] - x[i]^2) for i = 1:n-1]; [x[i] - 1 for i = 1:n-1]], (1:n) ./ (n + 1), 2n - 2)
+
+    stats = with_logger(NullLogger()) do
+      solver(nls, max_time=600.0, max_eval=-1)
```
Comment (dpo) [inline test/solvers/unconstrained-nls.jl]: Why allow 10 minutes for this?

File: test/solvers/unconstrained-nls.jl [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+      stats = with_logger(NullLogger()) do
+        solver(nls, max_eval=-1)
+      end
+      @test isapprox(stats.solution, ones(2), atol=1e-6)
+      @test isapprox(stats.objective, 0.0, atol=1e-6)
+      @test stats.dual_feas < 1e-6
+      @test stats.status == :first_order
+    end
+
+    @testset "Underdeterminded" begin
```
Comment (dpo) [inline test/solvers/unconstrained-nls.jl]: Determinded.

File: test/solvers/unconstrained.jl (hunk 1 above)
Comment (abelsiqueira) [reply test/solvers/unconstrained.jl]: Nothing really. It's legacy code. I think we can remove the `CUTEst` problems everywhere.

File: test/solvers/unconstrained-nls.jl (hunk 2 above)
Comment (abelsiqueira) [reply test/solvers/unconstrained-nls.jl]: Can't remember. It was in TRON, but it doesn't need that much time (I may have decreased `n` at some point).

File: test/solvers/unconstrained-nls.jl (hunk 3 above)
Comment (abelsiqueira) [reply test/solvers/unconstrained-nls.jl]: In this case, `m < n` and there are infinite solutions. Shouldn't be `underdetermined`?

File: test/solvers/unconstrained.jl (hunk 1 above)
Comment (dpo) [reply test/solvers/unconstrained.jl]: i agree.

File: test/solvers/unconstrained-nls.jl (hunk 2 above)
Comment (dpo) [reply test/solvers/unconstrained-nls.jl]: 10 minutes is excessive here.

File: test/solvers/unconstrained-nls.jl (hunk 3 above)
Comment (dpo) [reply test/solvers/unconstrained-nls.jl]: I mean that you wrote "determinded" instead of "determined".

Comment (dpo) [reply test/solvers/unconstrained-nls.jl]: (And so did I, I now realize... sorry)

Comment (abelsiqueira) [reply test/solvers/unconstrained-nls.jl]: Ah, sorry. Will fix

File: test/solvers/unconstrained.jl (hunk 1 above)
Comment (abelsiqueira) [reply test/solvers/unconstrained.jl]: There is CUTEst test in [trunkls](https://github.com/JuliaSmoothOptimizers/JSOSolvers.jl/pull/4/files#diff-c0dce8801f06ebb9221fa66f46546db8R18). Should I remove it, or do you have a suggestion on how to change it?

Comment (dpo) [reply test/solvers/unconstrained.jl]: I would replace it with a ADNLSModel.

**Comment by abelsiqueira** [comment]
Updated

**Comment by dpo** [comment]
Thank you!

**Pull request merged**